src/components/transactions/Repay/CollateralRepayActions.tsx

```diff
@@ -85,7 +85,9 @@ export const CollateralRepayActions = ({
       sx={sx}
       {...props}
       handleAction={action}
-      handleApproval={() => approval()}
+      handleApproval={() =>
+        approval({ amount: repayAmount, underlyingAsset: poolReserve.aTokenAddress })
```

Comment:
Need to test this to figure out why the current version is working with blank parameters

Reply:
The reason this works is that these params are only used by permit, added them anyways so it will support in the future

Reply:
Would it clean it up if we had a `permit` function and an `approval` function in `useTransactionHandler`? That way it'd be clear when/why these parameters are needed.